xclim/subset.py

```diff
@@ -43,14 +45,58 @@ def func_checker(*args, **kwargs):
     return func_checker
 
 
+def check_start_end_dates(func):
+    def func_checker(*args, **kwargs):
+        """
+        A decorator to verify that start and end dates are valid in a time subsetting function.
+        """
+        da = args[0]
+        if "start_date" not in kwargs:
+            # use string for first year only - .sel() will include all time steps
+            kwargs["start_date"] = da.time.min().dt.strftime("%Y").values
+        if "end_date" not in kwargs:
+            # use string for last year only - .sel() will include all time steps
+            kwargs["end_date"] = da.time.max().dt.strftime("%Y").values
+
+        try:
+            da.time.sel(time=kwargs["start_date"])
+        except KeyError:
+            warnings.warn(
+                '"start_date" not found within input date time range. Defaulting to minimum time step in '
+                "xarray object.",
+                Warning,
+                stacklevel=2,
+            )
+            kwargs["start_date"] = da.time.min().dt.strftime("%Y").values
+        try:
+            da.time.sel(time=kwargs["end_date"])
+        except KeyError:
+            warnings.warn(
+                '"end_date" not found within input date time range. Defaulting to maximum time step in '
+                "xarray object.",
+                Warning,
+                stacklevel=2,
+            )
+            kwargs["end_date"] = da.time.max().dt.strftime("%Y").values
+
+        if (
+            da.time.sel(time=kwargs["start_date"]).min()
+            > da.time.sel(time=kwargs["end_date"]).max()
+        ):
+            raise ValueError("Start date is after end date.")
+
+        return func(*args, **kwargs)
+
+    return func_checker
+
+
 def check_lons(func):
     def func_checker(*args, **kwargs):
         """
         A decorator to reformat user-specified lon values based on the lon dimensions of a supplied xarray
```

Comment:
I think the docstring could be more explicit about what is done in this decorator.

Reply:
Done.